.changesets/fix_garypen_router_243_batch_fetch_filter.md

```diff
@@ -0,0 +1,9 @@
+### Filter the fetches we add to a batch when we create the batch ([PR #5034](https://github.com/apollographql/router/pull/5034))
+
+Without filtering the query hashes during batch creation we could end up in a situation where we have additional query hashes in the batch.
+
+This manifests itself by batch queries failing with query hashes appearing as committed without ever having been registered in a batch.
+
+Filtering during batch creation is now matching the filtering at subgraph coordination and fixes this issue.
```

Comment:
```suggestion
### Filter fetches added to batch during batch creation ([PR #5034](https://github.com/apollographql/router/pull/5034))

Previously, the router didn't filter query hashes when creating batches. This could result in failed queries because the additional hashes could incorrectly make a query appear to be committed when it wasn't actually registered in a batch. 

This release fixes this issue by filtering query hashes during batch creation.
```